File: js/sorting-list.js
const array = [1, 123, 22, 4, 51, 3, 5, 60, 8, 53, 2, 10, 11, 15, 9, 16];

const arrayEven = array.filter(value => !(value % 2)).sort((a, b) => b - a);
const arraySort = [];

array.forEach(value => {
	if (value % 2) {
		arraySort.push(value);
	} else {
		arraySort.push(arrayEven.pop());
	}
});

console.log(arraySort);

File: lib/sorting-list.dart
final List<int> array = [10, 5, 2, 14, 11, 12, 3, 4, 8, 18, 33, 7, 50, 60, 16];

List<int> arrayEven = array.where((element) => element.isEven).toList();
List<int> arraySorted = [];

void answer() {

  arrayEven.sort((a, b) => b - a);

  array.forEach((element) {
    if (element % 2 == 0) {
      arraySorted.add(arrayEven.removeLast());
    } else {
      arraySorted.add(element);
    }
  });

	print(array);
  print(arraySorted);
}
